src/nri/sampler.go

```diff
@@ -153,14 +153,35 @@ func attributes(container types.Container) []entry {
 			cname = cname[1:]
 		}
 	}
-	return []entry{
+	entries := []entry{
 		metricCommandLine(container.Command),
 		metricContainerName(cname),
 		metricContainerImage(container.ImageID),
 		metricContainerImageName(container.Image),
 		metricState(container.State),
 		metricStatus(container.Status),
 	}
+
+	// Removes attributes with emtpy values to avoid be reported.
+	sanitizedEntries := []entry{}
+	for _, entry := range entries {
+		if isAttributeValueEmpty(entry) {
+			continue
+		}
+
+		sanitizedEntries = append(sanitizedEntries, entry)
+	}
+
+	return sanitizedEntries
+}
+
+func isAttributeValueEmpty(e entry) bool {
```

Comment:
can we check if this would affect possibly empty values (if supported)

Reply:
I added some assert to the tests to check that labels are still being reported when they have empty values which is supported by docker.